### ISSUE TITLE
No event information in download.ics

### ISSUE THREAD

**Issue by apogorzelska** [comatory/fb2iCal#42]
Service at https://fb2ical-3051b.web.app/ returns following for event https://www.facebook.com/events/351382967310859, which doesn't contain actual event information:

```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
PRODID:adamgibbons/ics
METHOD:PUBLISH
X-PUBLISHED-TTL:PT1H
BEGIN:VEVENT
UID:f1027b76-9b78-41b3-8604-7c5e73b7d153
SUMMARY:Facebook
DTSTAMP:20231104T151601Z
DTSTART:20231104T151900Z
URL:https://www.facebook.com/
DURATION:PT120M
END:VEVENT
END:VCALENDAR
```

**Comment by comatory** [comment]
Thank you for reporting it. I will look into it why that happens.

**Comment by comatory** [comment]
@apogorzelska I tried converting your event. The code in the application is kind of old and it relies on HTML structure (either presence of `ld+json` tag or DOM) which seems like it's not compatible anymore.
I will see if I can fix it, nowadays it'd make more sense to use some ML-based parsing instead.

**Comment by cjduncana** [comment]
I found another example of this happening. I tried importing https://www.facebook.com/events/3897788000484045/3897788007150711/, but the ical file doesn't have the correct information.

```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
PRODID:adamgibbons/ics
METHOD:PUBLISH
X-PUBLISHED-TTL:PT1H
BEGIN:VEVENT
UID:b2a0e671-2762-4ec7-a9d8-8fb0543eec95
SUMMARY:Error
DTSTAMP:20250707T145105Z
DTSTART:20250707T105100Z
URL:https://mobile.facebook.com/events/3897788000484045/3897788007150711/
DURATION:PT120M
END:VEVENT
END:VCALENDAR
```

**Comment by comatory** [comment]
@cjduncana I'm surprised somebody is using this application at all 😅 Do you use it regularly or was it just one off thing?

I'm thinking about resuming work on this.

**Comment by cjduncana** [comment]
@comatory 
It was a one-off 😅

**Comment by comatory** [comment]
Yeah the thing is, FB is constantly changing the web app and this app is relying on certain things being available, like the LD+JSON inside of `meta` tag. When it's not there, it's not going to give good results, unfortunately.
